### PR TITLE
GCP Certification Testing Infrastructure

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description

Adds GCP Authenticating via Workload Identity Federation.

The GCP resources (Authentication/Authorization) have been already deployed to the Dapr GCP Project that will host the Certification Tests resources.
Here is the Dapr GCP Project: https://console.cloud.google.com/home/dashboard?project=dapr-tests

This workflow changes do not require secrets to gain access to GCP. The Authn/Authz is done through the use of the [GCP Workload Identity Federation](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation) which is handled by 2 resources already deployed (through Terraform) in the GCP Project:

- Workload Identity Pool & Workload Identity Provider Configured **specifically** for this Github repository -  **(Authn)**
- A GCP IAM Service Account (SA) used to impersonate this Github Actions workflow within the GCP Project **(Authz)**.  This SA has been assigned the roles **`roles/pubsub.admin`** and **`roles/datastore.owner`** which will be used for the `GCP PusbSub` and `GCP Firestore` Certification Tests.

For further information of how the Authn/Authz GCP <==> Github Actions works, see the [google-github-actions/auth](https://github.com/google-github-actions/auth/blob/main/README.md)

## Issue reference

PR will close: #2651

### CODE REVIEW

**Comment by berndverst** [comment]
Today we don't use the the test/infra repo in contrib at all. Can you explain why this is done? This will make it difficult for contrib maintainers to maintain things.

**Comment by robertojrojas** [comment]
@berndverst I've updated the PR description with more information. The GCP resources needed by this PR have been applied so it should work as it stands. I'll submit a PR to this repo with the Terraform code used to generate those resources.

File: .github/scripts/test-info.mjs [hunk 1]
```diff
@@ -577,6 +580,9 @@ function GenerateMatrix(testKind, enableCloudTests) {
             'require-aws-credentials': comp.requireAWSCredentials
                 ? 'true'
                 : undefined,
+                'require-gcp-credentials': comp.requireGCPCredentials
```
Comment (ItalyPaleAle) [inline .github/scripts/test-info.mjs:583]:
```suggestion
            'require-gcp-credentials': comp.requireGCPCredentials
```

File: .github/workflows/certification.yml [hunk 2]
```diff
@@ -159,6 +166,22 @@ jobs:
...
+    #    - GCP_PROJECT
+    #    - GOOGLE_APPLICATION_CREDENTIALS
+    - id: 'auth'
+      if: matrix.require-gcp-credentials == 'true'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        token_format: 'access_token'
+        workload_identity_provider: 'projects/369878874207/locations/global/workloadIdentityPools/contrib-cert-tests-gh-pool/providers/contrib-cert-tests-gh-provider'
+        service_account: 'comp-contrib-wif@dapr-tests.iam.gserviceaccount.com'
```
Comment (ItalyPaleAle) [inline .github/workflows/certification.yml:180]: I think these should be defined as a variable in the project. @berndverst and I can help you add that to the repo

File: .github/workflows/conformance.yml [hunk 3]
```diff
@@ -155,6 +162,22 @@ jobs:
...
+    #    - GCP_PROJECT
+    #    - GOOGLE_APPLICATION_CREDENTIALS
+    - id: 'auth'
+      if: matrix.require-gcp-credentials == 'true'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        token_format: 'access_token'
+        workload_identity_provider: 'projects/369878874207/locations/global/workloadIdentityPools/contrib-cert-tests-gh-pool/providers/contrib-cert-tests-gh-provider'
+        service_account: 'comp-contrib-wif@dapr-tests.iam.gserviceaccount.com'
```
Comment (ItalyPaleAle) [inline .github/workflows/conformance.yml:176]: Here too. Please, let's make these variables.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: .github/workflows/certification.yml (hunk 2 above)
Comment (robertojrojas) [reply .github/workflows/certification.yml:180]: @ItalyPaleAle  @berndverst 💯 Agreed! I was thinking about these variables:
```
 workload_identity_provider: ${{ secrets.GCP_WIF_PROVIDER_NAME }}
 service_account: ${{ secrets.GCP_WIF_SA_EMAIL }}
```
Does that make sense?

Comment (robertojrojas) [reply .github/workflows/certification.yml:180]: Here are the values for the SECRETS vars: 

- `secrets.GCP_WIF_PROVIDER_NAME`=**'projects/369878874207/locations/global/workloadIdentityPools/contrib-cert-tests-gh-pool/providers/contrib-cert-tests-gh-provider'**
- `secrets.GCP_WIF_SA_EMAIL`=**'comp-contrib-wif@dapr-tests.iam.gserviceaccount.com'**

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**